From b358f507968bd9db3166aa45afe11a33e40c7337 Mon Sep 17 00:00:00 2001
From: denglei <denglei@eswincomputing.com>
Date: Tue, 29 Oct 2024 09:41:49 +0800
Subject: [PATCH 277/654] WIN2030-16445:fix:Resolve DSP & NPU driver
 compilation issue.

Changelogs:
	1.Resolve DSP & NPU driver compilation issue.
	2.Default opening of NPU & DSP driver module.

Change-Id: Ifc0541a3d8c2f25fc44f8c75ca629b1d670425e9
Signed-off-by: denglei <denglei@eswincomputing.com>
---
 arch/riscv/configs/eic7702_defconfig                      | 2 ++
 drivers/soc/eswin/ai_driver/common/devices/mailbox_regs.h | 2 ++
 drivers/soc/eswin/ai_driver/common/hetero_ipc.h           | 2 ++
 drivers/soc/eswin/ai_driver/dsp/Makefile                  | 1 -
 drivers/soc/eswin/ai_driver/npu/Makefile                  | 1 -
 drivers/soc/eswin/ai_driver/npu/npu_spram.c               | 1 +
 6 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/arch/riscv/configs/eic7702_defconfig b/arch/riscv/configs/eic7702_defconfig
index 9318fc21eaa9..5ae50b34eea1 100644
--- a/arch/riscv/configs/eic7702_defconfig
+++ b/arch/riscv/configs/eic7702_defconfig
@@ -742,6 +742,8 @@ CONFIG_ARM_SMMU_V3=y
 CONFIG_RPMSG_VIRTIO=y
 CONFIG_ARCH_ESWIN_EIC770X_SOC_FAMILY=y
 CONFIG_ARCH_ESWIN_EIC7702_SOC=y
+CONFIG_ESWIN_DSP=m
+CONFIG_ESWIN_NPU=m
 CONFIG_EIC7X_D2D=y
 CONFIG_EXTCON=y
 CONFIG_MEMORY=y
diff --git a/drivers/soc/eswin/ai_driver/common/devices/mailbox_regs.h b/drivers/soc/eswin/ai_driver/common/devices/mailbox_regs.h
index 98558ecf0af5..f4be79fac599 100644
--- a/drivers/soc/eswin/ai_driver/common/devices/mailbox_regs.h
+++ b/drivers/soc/eswin/ai_driver/common/devices/mailbox_regs.h
@@ -202,10 +202,12 @@
 
 #define ESWIN_MAILBOX_NPU_LOCK_BIT BIT3
 
+#ifndef __KERNEL__
 static u32 MAILBOX_E31_TO_DSP_REG[] = {ESWIN_MAILBOX_U84_TO_DSP_0_REG_BASE, ESWIN_MAILBOX_U84_TO_DSP_1_REG_BASE,
                                        ESWIN_MAILBOX_U84_TO_DSP_2_REG_BASE, ESWIN_MAILBOX_U84_TO_DSP_3_REG_BASE};
 
 static u32 MAILBOX_E31_TO_DSP_INT[] = {MBOX_INT_TO_DSP0, MBOX_INT_TO_DSP1, MBOX_INT_TO_DSP2, MBOX_INT_TO_DSP3};
+#endif
 
 #define MBOX_INT_BIT_BASE 5
 
diff --git a/drivers/soc/eswin/ai_driver/common/hetero_ipc.h b/drivers/soc/eswin/ai_driver/common/hetero_ipc.h
index c255d5cd371b..667e13996be3 100644
--- a/drivers/soc/eswin/ai_driver/common/hetero_ipc.h
+++ b/drivers/soc/eswin/ai_driver/common/hetero_ipc.h
@@ -444,6 +444,7 @@ static void messagebox_send(u16 channel_id, msg_payload_t payload)
     reg_write((size_t)(mbox_base + MBOX_NPU_WR_DATA1_OFFSET), MBOX_WRITE_FIFO_BIT);
 }
 
+#ifndef __KERNEL__
 /**
  *  @brief Send a mailbox message payload to DSP. This function should be used
  *  by E31.
@@ -488,6 +489,7 @@ static void messagebox_send_dsp(u32 op_type, u64 payload)
     // clear lock bit
     reg_write(mbox_base + MBOX_NPU_WR_LOCK, 0);
 }
+#endif
 
 /**
  * @brief Receives data from a mailbox message.
diff --git a/drivers/soc/eswin/ai_driver/dsp/Makefile b/drivers/soc/eswin/ai_driver/dsp/Makefile
index ebacf5f1723e..e433e1af883b 100755
--- a/drivers/soc/eswin/ai_driver/dsp/Makefile
+++ b/drivers/soc/eswin/ai_driver/dsp/Makefile
@@ -30,7 +30,6 @@ DEBUG_LEVEL ?= 0
 ccflags-y := -I$(src)/
 ccflags-y += -I$(srctree)/drivers/soc/eswin
 ccflags-y += -I$(srctree)/drivers/memory/eswin/es_dev_buf/include/linux
-ccflags-y += -I$(srctree)/drivers/memory/eswin/es_iommu_rsv/include/linux
 ccflags-y += -I$(srctree)/$(src)/../common
 ccflags-y += -I$(srctree)/$(src)/../common/devices
 
diff --git a/drivers/soc/eswin/ai_driver/npu/Makefile b/drivers/soc/eswin/ai_driver/npu/Makefile
index 3602d3c9a0f6..29cbe78dd026 100644
--- a/drivers/soc/eswin/ai_driver/npu/Makefile
+++ b/drivers/soc/eswin/ai_driver/npu/Makefile
@@ -10,7 +10,6 @@ ccflags-y += -I$(srctree)/$(src)/include
 ccflags-y += -I$(srctree)/$(src)/../dsp
 ccflags-y += -I$(srctree)/$(src)/../common
 ccflags-y += -I$(srctree)/$(src)/../common/devices
-ccflags-y += -I$(srctree)/drivers/memory/eswin/es_iommu_rsv/include/linux
 ccflags-y += -I$(srctree)/drivers/memory/eswin/es_dev_buf/include/linux
 
 ccflags-y += -DNPU_DEV_SIM=$(NPU_DEV_SIM)
diff --git a/drivers/soc/eswin/ai_driver/npu/npu_spram.c b/drivers/soc/eswin/ai_driver/npu/npu_spram.c
index 02a17f2b892c..2e8b78a88e73 100644
--- a/drivers/soc/eswin/ai_driver/npu/npu_spram.c
+++ b/drivers/soc/eswin/ai_driver/npu/npu_spram.c
@@ -27,6 +27,7 @@
 #include <linux/delay.h>
 #include <linux/uaccess.h>
 #include "npu_spram.h"
+#include "npu_base_regs.h"
 
 // At least one way must be reserved for cache
 #define MAX_CACHE_SIZE (2 * 0x100000)  //2MB
-- 
2.51.0

